### PR TITLE
Preserve input dtype in population code helpers

### LINKED ISSUE #429: snnTorch population-coded accuracy can mis-score close float64 predictions
## Summary

I found a reproducible issue in `snntorch.functional.acc.accuracy_rate` when using `population_code=True`. For close `float64` population-coded class scores, the public accuracy helper can report the wrong class even though the target class has the larger score.

## Minimal Reproduction

```python
import torch
from snntorch.functional import acc

spk_out = torch.tensor([[[1.00000002, 1.00000005]]], dtype=torch.float64)
targets = torch.tensor([1])

accuracy = acc.accuracy_rate(
    spk_out,
    targets,
    population_code=True,
    num_classes=2,
)
pop_code = acc._population_code(spk_out, num_classes=2, num_outputs=2)

print("accuracy:", accuracy)
print("population_code:", pop_code)

assert accuracy == 1.0
```

## Expected Behavior

The population-coded metric should preserve the class ordering implied by the input scores, or explicitly document/cast precision in a way that does not silently change the predicted class.

## Actual Behavior

The assertion fails. In my reproduction, the helper returned:

```text
accuracy: 0.0
population_code: tensor([[1., 1.]])
```

The close `float64` score difference is lost, so the metric reports the wrong result.

### PR DESCRIPTION
**Pull request by rlogger**
Fixes #429, and the `ce_count_loss` float64 case from #428.

`_population_code` in both `snntorch/functional/acc.py` and `snntorch/functional/loss.py` allocates its accumulator with `torch.zeros(...)`, which always produces float32 regardless of the input dtype. This causes two problems with float64 inputs:

1. **Silent precision loss in `accuracy_rate`**: close float64 scores are downcast to float32, which can collapse them to equal values and flip the predicted class (the reproduction in #429 reports 0.0 accuracy when the target class has the strictly larger score).
2. **`RuntimeError` in `ce_count_loss(population_code=True)`**: the float32 population code is passed to `NLLLoss` alongside float64 class weights, raising `expected scalar type Float but found Double`.

The fix allocates the accumulator with `dtype=spk_out.dtype` (and passes `device=` directly instead of the extra `.to(device)` copy) so the population code always matches the input.

Added regression tests for both cases:
- `test_accuracy_rate_population_code_dtype`: checks that close float64 scores keep their ordering and the accuracy is scored correctly, plus a couple of basic `accuracy_rate` tests since `acc.py` had no coverage before.
- `test_ce_count_loss_population_code_dtype`: checks that population-coded `ce_count_loss` works with float64 inputs/weights and returns a float64 loss.

Full test suite passes locally (150 passed + NIR tests), and the changed files are formatted with black. float32 behavior is unchanged.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by ixfd64 — APPROVED** [review]
Looks fine to me.
